CMakeLists.txt

```diff
@@ -358,6 +358,7 @@ if(MSVC)
     set(${flag} "${${flag}} /wd4800") # forcing value to bool 'true' or 'false' (performance warning)
     set(${flag} "${${flag}} /wd4996") # The POSIX name for this item is deprecated.
   endforeach()
+  add_compile_options("/Zc:__cplusplus")
```

Comment:
i added this quickly as a test, you probably want to add a comment here to explain what it does/why it's needed. note that this is (i think) going to set `__cplusplus` to `201402L`, there is no "C++11" mode for MSVC.

Reply:
The only problem with this is that I have no idea what it does ;) From your exchanges with James I understand it has to do with how the compiler treats `NULL`, but I don't understand much more than that. Do you have a suggestion for the wording here?

Reply:
"this flag causes MSVC to correctly report `__cplusplus`, which prevents a compiler error caused by some versions of libsndfile's C++ header redefining nullptr as NULL. See https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/"

Reply:
that's how I'd word it. it's not so much that it changes how MSVC treats nullptr or null, it affects the C++ standard conformance macro `__cplusplus`, which triggers code in the libsndfile header that does `#define nullptr NULL`. since preprocessing happens before compilation, this means all later instances of `nullptr` in the same translation unit are replaced with `NULL`, which in turn typically expands to 0. IIUC normally this is not a problem as whatever NULL expands to is required to be implicitly convertible to a null pointer value, but I guess in this case it makes a certain function template's template type arguments impossible to deduce.